skxray/speckle_visibility/speckle_visibility.py

```diff
@@ -0,0 +1,293 @@
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+    This module will provide statistics analysis for the speckle pattern
+    to use in X-ray Speckle Visibility Spectroscopy (XSVS)
+"""
+
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+from six.moves import zip
+from six import string_types
+
+import skxray.correlation as corr
+import skxray.roi as roi
+import skxray.core as core
+
+try:
+    iteritems = dict.iteritems
+except AttributeError:
+    iteritems = dict.items  # python 3
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def max_counts(sample_dict, label_array):
+    """
+    This will determine the highest speckle counts occurred in the required
+    ROI's in required images.
+
+    Parameters
+    ----------
+    sample_dict : dict
+        sets of images as a dictionary
+
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    max_counts : int
+        maximum speckle counts
+    """
+    max_cts = 0
+    for key, img_sets in iteritems(sample_dict):
+        for n, img in enumerate(img_sets.operands[0]):
+            int_dist = intensity_distribution(img, label_array)
+            for j in range(len(int_dist)):
+                counts = np.max(list(int_dist.values())[j])
+                if max_cts < counts:
+                    max_cts = counts
+    return max_cts
+
+
+def intensity_distribution(image_array, label_array):
+    """
+    This will provide the intensity distribution of the ROI"s
+    eg: radial intensity distributions of a
+    rings of the label array
+    Parameters
+    ----------
+    image_array : array
+        image data dimensions are: (rr, cc)
+
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    radial_intensity : dict
+        radial intensity of each ROI's
+    """
+
+    if label_array.shape != image_array.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+
+    labels, indices = corr.extract_label_indices(label_array)
+    label_num = np.unique(labels)
+
+    intensity_distribution = {}
+
+    for n in label_num:
+        value = (np.ravel(image_array)[indices[labels == n].tolist()])
+        intensity_distribution[n] = value
+
+    return intensity_distribution
+
+
+def time_bining(number=2, number_of_images=50):
+    """
+    This will provide the time binning for the integration.
```

Comment:
although, it looks like logs are really expensive and the existing version is faster....